addons/wowl/static/src/services/ui/ui.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, core, tags, useState } from "@odoo/owl";
+import type { OdooEnv, Service } from "../../types";
+
+const { EventBus } = core;
+
+export interface UIService {
+  block: () => void;
+  unblock: () => void;
+}
+
+class BlockUI extends Component<{}, OdooEnv> {
+  static template = tags.xml`
+    <div>
+      <div t-if="state.blockUI" class="o_blockUI" >
+        <div class="o_spinner">
+            <img src="/wowl/static/img/spin.png" alt="Loading..."/>
+        </div>
+        <div class="o_message">
+            <t t-raw="state.message"/>
+        </div>
+      </div>
+    </div>`;
+  messagesByDuration = [
+    { time: 20, text: "Loading..." },
+    { time: 40, text: "Still loading..." },
+    { time: 60, text: "Still loading...<br />Please be patient." },
+    { time: 180, text: "Don't leave yet,<br />it's still loading..." },
+    { time: 120, text: "You may not believe it,but the application is actually loading..." },
+    { time: 3180, text: "Take a minute to get a coffee,<br />because it's loading..." },
+    { time: 7200, text: "Maybe you should consider reloading the application by pressing F5..." },
```

Comment:
use Number.MAX_VALUE ? or set it to null, and in this case do not set a timeout?